### PR TITLE
Rename variables and functions & add checks and tests

### PR DESCRIPTION
**Pull request by garyghayrat**
- Added check for mint price is less then or equal to max price

- Added Emit minting event that includes receiving address, token Id, AND payment amount for mint

- Added a version of `payAndMint` that takes the` _mintTo` address and refactored w/ shared internal method

### CODE REVIEW

**Review by apbendi — COMMENTED** [review]
Some comments @garyghayrat. Thanks!

File: src/CreatorToken.sol [hunk 1]
```diff
@@ -23,16 +27,19 @@ contract CreatorToken is ERC721 {
 
   function payAndMint(uint256 _maxPayment) public {
     uint256 _mintPrice = _temporaryGetNextTokenPrice();
+    if (_mintPrice > _maxPayment) revert CreatorToken__MaxPaymentExceeded(_mintPrice, _maxPayment);
+    _mintAndTransfer(msg.sender, _mintPrice);
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply src/CreatorToken.sol]: Makes sense, updated here: https://github.com/showtime-xyz/creator-tokens/pull/5/commits/35fac7af12976e042d61dd446eee665df3068ebe

File: src/CreatorToken.sol [hunk 2]
```diff
@@ -23,16 +27,19 @@ contract CreatorToken is ERC721 {
...
 
-    // TODO: Validate mint price is less then or equal to max price
-
-    // TODO: Emit minting event that includes receiving address, token Id, AND payment amount for
-    // mint
-
-    // TODO: Write a version of this method that takes the _mintTo address (refactor w/ shared
-    // internal method)
+  function payAndMint(address _mintTo, uint256 _maxPayment) public {
+    uint256 _mintPrice = _temporaryGetNextTokenPrice();
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply src/CreatorToken.sol]: Updated in the same commit mentioned above: https://github.com/showtime-xyz/creator-tokens/pull/5/commits/35fac7af12976e042d61dd446eee665df3068ebe

File: src/CreatorToken.sol [hunk 3]
```diff
@@ -23,16 +27,19 @@ contract CreatorToken is ERC721 {
...
+    if (_mintPrice > _maxPayment) revert CreatorToken__MaxPaymentExceeded(_mintPrice, _maxPayment);
+    _mintAndTransfer(_mintTo, _mintPrice);
+  }
 
-    _mintAndIncrement(msg.sender);
+  function _mintAndTransfer(address _mintTo, uint256 _mintPrice) internal {
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply src/CreatorToken.sol]: Updated in the same commit mentioned above: https://github.com/showtime-xyz/creator-tokens/commit/35fac7af12976e042d61dd446eee665df3068ebe

File: test/CreatorToken.t.sol [hunk 4]
```diff
@@ -48,8 +51,50 @@ contract Minting is CreatorTokenTest {
     creatorToken.payAndMint(BASE_PAY_AMOUNT);
     vm.stopPrank();
 
-    assertEq(creatorToken.balanceOf(_minter), 1);
+    assertEq(creatorToken.balanceOf(_minter), originalMinterBalance + 1);
+    assertEq(payToken.balanceOf(_minter), 0);
+    assertEq(payToken.balanceOf(address(creatorToken)), BASE_PAY_AMOUNT);
+  }
+
+  function test_payAndMintWithMintToAddress(address _minter, address _mintTo) public {
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply test/CreatorToken.t.sol:59]: yes!

File: test/CreatorToken.t.sol [hunk 5]
```diff
@@ -48,8 +51,50 @@ contract Minting is CreatorTokenTest {
...
+    vm.stopPrank();
+
+    assertEq(creatorToken.balanceOf(_mintTo), originalMinterBalance + 1);
     assertEq(payToken.balanceOf(_minter), 0);
     assertEq(payToken.balanceOf(address(creatorToken)), BASE_PAY_AMOUNT);
   }
+
+  function test_EmitsMintedEvent(address _minter) public {
+    uint256 lastIdStorageSlot = 8; // from `forge inspect CreatorToken storage-layout` command
+    uint256 tokenId = uint256(vm.load(address(creatorToken), bytes32(lastIdStorageSlot))) + 1;
```
[the remark this answers is not in the document]
Comment (garyghayrat) [reply test/CreatorToken.t.sol:78]: Yea, agreed.

File: test/CreatorToken.t.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (garyghayrat) [reply test/CreatorToken.t.sol]: Update here: https://github.com/showtime-xyz/creator-tokens/pull/5/commits/138b245337eac98154bcbf241d7196c043903221

File: test/CreatorToken.t.sol (hunk 5 above)
[the remark this answers is not in the document]
Comment (garyghayrat) [reply test/CreatorToken.t.sol]: Commit: https://github.com/showtime-xyz/creator-tokens/pull/5/commits/138b245337eac98154bcbf241d7196c043903221

**Review by apbendi — APPROVED** [review]
Looks good, thanks @garyghayrat

**Pull request merged**